src/components/search-form/SearchForm.js

```diff
@@ -71,12 +71,16 @@ class SearchForm extends React.Component<Props, State> {
         isEmpty: true,
     };
 
-    UNSAFE_componentWillReceiveProps(nextProps) {
-        const { value } = nextProps;
-        // update state if input is controlled and becomes empty
-        if (typeof value !== 'undefined' && !value.trim()) {
-            this.setState({ isEmpty: true });
+    static getDerivedStateFromProps(props, state) {
+        const { value } = props;
+
+        if (value !== state.value && value !== 'undefined' && !value.trim()) {
+            return {
```

Comment:
I realize it would be a larger change, but can we remove the `isEmpty` state property entirely and derive it based on `value` inside the render method?

Reply:
It would be a larger a change, and would just be shifting things around in the component. We would make a it a fully controlled component which would be an API change, or we need to track the value or something to trigger re-renders in the component. If we want to make these changes we may want to wait for another PR as this is concerned with migration of lifecycle methods.